File: bms/go/go-edlib/OrderedMap_Swap/OrderedMap_Swap.go
package test

var ()

const ()

func (p OrderedMap) Swap(i, j int) { p[i], p[j] = p[j], p[i] }

type pair struct {
	Key   string
	Value float32
}

// OrderedMap is a slice of pairs type with string keys and float values.
// It implement sorting methods by values.
type OrderedMap []pair

File: translations_only/fail/go/go-edlib/OrderedMap_Swap/gpt4/CAPR/OrderedMap_Swap.rs

struct Pair {
    key: String,
    value: f32,
}

type OrderedMap = Vec<Pair>;

fn swap(ordered_map: &mut OrderedMap, i: isize, j: isize) {
    let len = ordered_map.len() as isize;
    if i >= 0 && j >= 0 && i < len && j < len {
        ordered_map.swap(i as usize, j as usize);
    }
}
